core/types/smt/byte32.go

```diff
@@ -0,0 +1,18 @@
+package smt
+
+import (
+	"github.com/iden3/go-iden3-crypto/poseidon"
+	"math/big"
+)
+
+type Byte32 [32]byte
+
+func (b *Byte32) Hash() (*big.Int, error) {
+	first16 := new(big.Int).SetBytes(b[0:16])
```

Comment:
Just use `new(big.Int).SetBytes(b)` is enough?

Reply:
no, need to seperate into two parts